editor/selectors.js

```diff
@@ -8,48 +8,121 @@ import { first, last, get } from 'lodash';
  */
 import { addQueryArgs } from './utils/url';
 
+/**
+ * Returns the current editing mode.
+ *
+ * @param  {Object} state Global application state
+ * @return {String}       Editing mode
+ */
 export function getEditorMode( state ) {
 	return state.mode;
 }
 
+/**
+ * Returns true if the editor sidebar panel is open, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether sidebar is open
+ */
 export function isEditorSidebarOpened( state ) {
 	return state.isSidebarOpened;
 }
 
+/**
+ * Returns true if the past editor history snapshots exist, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether undo history exists
+ */
 export function hasEditorUndo( state ) {
 	return state.editor.history.past.length > 0;
 }
 
+/**
+ * Returns true if the future editor history snapshots exist, or false
+ * otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether redo history exists
+ */
 export function hasEditorRedo( state ) {
 	return state.editor.history.future.length > 0;
 }
 
+/**
+ * Returns true if the currently edited post is yet to be saved, or false if
+ * the post has been saved.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether the post is new
+ */
 export function isEditedPostNew( state ) {
 	return ! state.currentPost.id;
 }
 
+/**
+ * Returns true if there are unsaved values for the current edit session, or
+ * false if the editing state matches the saved or new post.
+ *
+ * @param  {Object}  state Global application state
+ * @return {Boolean}       Whether unsaved values exist
+ */
 export function isEditedPostDirty( state ) {
 	return state.editor.dirty;
 }
 
+/**
+ * Returns the post currently being edited.
```

Comment:
Not necessarily an empty object - I imagine a new post would contain `status: 'draft'` and any other default values.  Previous discussion: https://github.com/WordPress/gutenberg/pull/848#issuecomment-302834261

Currently these default values are stored in `post-content.js` ([#](https://github.com/WordPress/gutenberg/blob/728c17761585efbbd31ebef870b64428a7c88861/post-content.js#L8), [#](https://github.com/WordPress/gutenberg/blob/728c17761585efbbd31ebef870b64428a7c88861/editor/index.js#L30)).

So perhaps we could improve the wording here as follows:

> Returns our representation of the saved post object.  For an existing post, this is the last known state of the post on the server; for a new post, this is an object containing any relevant default values.